thread.c

```diff
@@ -24,7 +24,7 @@
 #include <pthread.h>
 #include "thread.h"
 
-bool start_detached_thread(void (*thread_fnc)(void*), void *argument) {
```

Comment:
this was the fn-type before my patch (which we both seem to agree, that this is what PThread wants)